internal/controllers/device_data_controller.go

```diff
@@ -344,31 +344,30 @@ func (udc *UserDevicesController) QueryDeviceErrorCodes(c *fiber.Ctx) error {
 // @Tags        user-devices
 // @Success     200 {object} controllers.GetUserDevicesErrorCodeQueriesResponse
 // @Security    BearerAuth
-// @Router      /user/devices/error-codes [get]
+// @Router      /user/devices/{userDeviceID}/error-codes [get]
 func (udc *UserDevicesController) GetUserDevicesErrorCodeQueries(c *fiber.Ctx) error {
+	udi := c.Params("userDeviceID")
 	userID := helpers.GetUserID(c)
 
 	userDevices, err := models.UserDevices(
 		models.UserDeviceWhere.UserID.EQ(userID),
-		qm.Load(models.UserDeviceRels.ErrorCodeQueries, qm.OrderBy("created_at desc")),
+		models.UserDeviceWhere.ID.EQ(udi),
+		qm.Load(models.UserDeviceRels.ErrorCodeQueries, qm.OrderBy(models.ErrorCodeQueryColumns.CreatedAt+" DESC")),
 	).All(c.Context(), udc.DBS().Reader)
 	if err != nil {
 		return err
 	}
 
-	resp := map[string][]GetUserDevicesErrorCodeQueriesResponse{}
+	resp := []GetUserDevicesErrorCodeQueriesResponse{}
 
 	for _, userDevice := range userDevices {
```

Comment:
I think that we’re iterating over a list of vehicles for a one-vehicle endpoint should give us pause!